scripts/madrid/predictions_collapsed/Snakefile

```diff
@@ -0,0 +1,134 @@
+SITE_NAME = "HUF_maxN21000/"
+MADRID_DATASET_STD_PATH = os.path.join(MADRID_DATASET_TOP_PATH, MADRID_VERSION, SITE_NAME)
+
+rule all:
+    input:
+        os.path.join(MADRID_DATASET_STD_PATH, 'alignedfeatures_per_tstep.csv'),
+        os.path.join(MADRID_DATASET_STD_PATH, 'normfeatures_per_tstep.csv'),
+        os.path.join(MADRID_DATASET_STD_PATH, 'CollapsedFeaturesPerSequence.csv'),
+        os.path.join(MADRID_DATASET_STD_PATH, 'Spec_CollapsedFeaturesPerSequence.json'),	
+        os.path.join(MADRID_DATASET_STD_PATH, 'x_train.csv'),
+        os.path.join(MADRID_DATASET_STD_PATH, 'x_test.csv'),
+        os.path.join(MADRID_DATASET_STD_PATH, 'y_train.csv'),
+        os.path.join(MADRID_DATASET_STD_PATH, 'y_test.csv')
+
+rule align_to_grid:
+    input:
+        align_script=os.path.join(PROJECT_REPO_DIR, 'src', 'align_to_grid.py'),
+        x_csv=os.path.join(MADRID_DATASET_STD_PATH, 'vitals.csv'),
+        x_spec_json=os.path.join(MADRID_DATASET_STD_PATH, 'Spec-Vitals.json')
+
+    output:
+        aligned_csv=os.path.join(MADRID_DATASET_STD_PATH, 'alignedfeatures_per_tstep.csv')
+
+
+    conda:
+        PROJECT_CONDA_ENV_YAML
+
+    shell:
+        '''
+        python -u {input.align_script} \
+            --input_ts_csv_path {input.x_csv} \
+            --data_dict {input.x_spec_json} \
+	    --step_size '1H' \
+            --output {output.aligned_csv} \
+        '''
+
+
+rule normalize_features:
+    input:
+        normalize_script=os.path.join(PROJECT_REPO_DIR, 'src', 'normalize_features.py'),
+        x_csv=os.path.join(MADRID_DATASET_STD_PATH, 'alignedfeatures_per_tstep.csv'),
+        x_spec_json=os.path.join(MADRID_DATASET_STD_PATH, 'Spec-Vitals.json')
+
+    output:
+        normx_csv=os.path.join(MADRID_DATASET_STD_PATH, 'normfeatures_per_tstep.csv')
+
+
+    conda:
+        PROJECT_CONDA_ENV_YAML
+
+    shell:
+        '''
+        python -u {input.normalize_script} \
+            --input {input.x_csv} \
+            --data_dict {input.x_spec_json} \
+            --output {output.normx_csv} \
+        '''
+
+
+
+rule collapse_features:
+    input:
+        script=os.path.join(PROJECT_REPO_DIR, 'src', 'feature_transformation.py'),
+        x_csv=os.path.join(MADRID_DATASET_STD_PATH, 'normfeatures_per_tstep.csv'),
+        x_spec_json=os.path.join(MADRID_DATASET_STD_PATH, 'Spec-Vitals.json')
+
+    output:
+        collapsedx_csv=os.path.join(MADRID_DATASET_STD_PATH, 'CollapsedFeaturesPerSequence.csv'),
+        collapsedx_json=os.path.join(MADRID_DATASET_STD_PATH, 'Spec_CollapsedFeaturesPerSequence.json')
+
+    conda:
+        PROJECT_CONDA_ENV_YAML
+
+    shell:
+        '''
+        python -u {input.script} \
+            --input {input.x_csv} \
+            --data_dict {input.x_spec_json} \
+            --output {output.collapsedx_csv} \
+            --data_dict_output {output.collapsedx_json} \
+            --range_pairs "[(0,100)]" \
+            --collapse
```

Comment:
This is where we want to make a change so we're creating a separate collapsed feature vector for each "prediction timestep" (e.g. every hour). That way we can properly evaluate a "risk score" rather than just a "given entire history, make one prediction" model

Probably want to be explicit about which summary features we are computing....